File: BEmu/BEmu/MarketDataRequest/ElementMarketSubscriptionStarted.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace BEmu.MarketDataRequest
{
    internal class ElementMarketSubscriptionStarted : Element
    {
        private readonly ElementMarketNull _exceptions;

        internal ElementMarketSubscriptionStarted(MessageMarketSubscriptionStarted arg)
        {
            this._exceptions = (ElementMarketNull)arg.Elements.First();
        }

        public override Name Name { get { return new Name("SubscriptionStarted"); } }
        public override int NumElements { get { return 1; } }
        public override int NumValues { get { return 1; } }
        public override bool IsArray { get { return false; } }
        public override bool IsComplexType { get { return true; } }
        public override bool IsNull { get { return false; } }

        public override object this[int index] { get { return null; } }
        public override Element this[string name]
        {
            get
            {
                if (name.ToUpper() == this._exceptions.Name.ToString().ToUpper())
                    return this._exceptions;
                else
                    return base[name];
            }
        }

        internal override StringBuilder PrettyPrint(int tabIndent)
        {
            return new StringBuilder();
        }

        public override string ToString()
        {
            return string.Format("SubscriptionStarted = {{{0}}}{0}", Environment.NewLine);
        }

    }
}

File: BEmu_cpp/src/MarketDataRequest/ElementMarketSubscriptionStarted.cpp
//------------------------------------------------------------------------------
// <copyright project="BEmu_cpp" file="src/MarketDataRequest/ElementMarketSubscriptionStarted.cpp" company="Jordan Robinson">
//     Copyright (c) 2013 Jordan Robinson. All rights reserved.
//
//     The use of this software is governed by the Microsoft Public License
//     which is included with this distribution.
// </copyright>
//------------------------------------------------------------------------------

#include "MarketDataRequest/ElementMarketSubscriptionStarted.h"
#include "MarketDataRequest/MessageMarketSubscriptionStarted.h"

namespace BEmu
{
	namespace MarketDataRequest
	{
		ElementMarketSubscriptionStarted::ElementMarketSubscriptionStarted(MessageMarketSubscriptionStarted * arg)
		{
			this->_exceptions = (ElementMarketNull*)arg->firstElement();
		}

		Name ElementMarketSubscriptionStarted::name() const
		{
			return Name("SubscriptionStarted");
		}

		size_t ElementMarketSubscriptionStarted::numValues() const
		{
			return 1;
		}

		size_t ElementMarketSubscriptionStarted::numElements() const
		{
			return 1;
		}

		bool ElementMarketSubscriptionStarted::isNull() const
		{
			return false;
		}

		bool ElementMarketSubscriptionStarted::isArray() const
		{
			return false;
		}

		bool ElementMarketSubscriptionStarted::isComplexType() const
		{
			return true;
		}

		std::ostream& ElementMarketSubscriptionStarted::print(std::ostream& stream, int level, int spacesPerLevel) const
		{
			stream << "SubscriptionStarted = {" << std::endl << '}' << std::endl;
			return stream;
		}
	}
}